src/model/docker.ts

```diff
@@ -111,9 +110,7 @@ class Docker {
                              steps and set the UNITY_LICENSE GitHub secret or enter a Unity
                              serial number inside the UNITY_SERIAL GitHub secret.`);
           }
-          unitySerial = this.getSerialFromLicenseFile(process.env.UNITY_LICENSE);
-        } else {
-          unitySerial = process.env.UNITY_SERIAL!;
+          process.env.UNITY_SERIAL = this.getSerialFromLicenseFile(process.env.UNITY_LICENSE);
```

Comment:
Note that it is not good practice to assign to environment from CI. GitHub has other ways to make it as masked.

Reply:
For some reason GitHub wasn't masking secrets in the calls. Even when I added the serial as a secret explicitly. Perhaps I was doing it wrong? This is more a bandaid solution though and am open to ways to fix it. It would be best to keep the serial from being required in the secrets as people with free licenses will need to do extra work to get the serial in there.

Reply:
Here's the [official docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-log) on how to achieve want you wanna do. 


Out of scope for this PR though. We need to absolutely keep things manageable when it comes to bugs.

Reply:
Just added setSecret to the latest commit and it seems to be working. Thanks for pointing me in the right direction!